<a href="https://colab.research.google.com/github/giovannacouto4-web/projeto1-/blob/main/Revis%C3%A3o_poly_versao_Giovanna.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Importe aqui os pacotes que achar necessários para a execução das questões
import pandas as pd
import altair as alt
import os
os.listdir()

['.config',
 'SATMAT.xls',
 'Censo_estados.xlsx',
 'FACCUSTO.xls',
 'TEMPOVIAGEMCN.xls',
 'TEMPOVIAGEMNY.xls',
 'CESSAFUMO.xls',
 'HABILITA.xls',
 'FORMATORADIO.xls',
 'FELICIDADE.xls',
 'OECD_IDH_data.csv',
 'EMISSAOCO2.xls',
 'InfoFamiliasEntrevistadas.csv',
 'NASCIDOSFORA.xls',
 'sample_data']

# Questão 1

### Classifique as variáveis abaixo segundo seus tipos

- (a) Quantidade de homicídios no Brasil em 2023.
- (b) Time de futebol.
- (c) Taxa de desemprego.
- (d) Nível de satisfação com o governo do Presidente Lula
- (e) Denominação religiosa
- (f) Renda per capita

1.
(a) quantitativa discreta
(b) qualitativa nominal
(c) quantitativa contínua
(d) qualitativa ordinal
(e) qualitativa nominal
(f) quantitativa contínua

# Questão 2

### Qual área de estudo?

Cerca de 1,5 milhão de estudantes se matricularam em faculdades e universidades em 2008.

No arquivo HABILITA.xls estão dados sobre o percentual de estudantes de primeiro ano que pretendiam se habilitar em diversas áreas disciplinares.

- (a) Quem eram, originalmente, as observações (unidades de análise) que geraram esses dados percentuais?
- (b) Verifique a consistência desses dados.
- (c) Faça uma análise desses dados a partir de gráficos e escreva uma breve interpretação dos resultados.







(a) Cada linha do arquivo representa uma porcentagem de estudantes, mas a observação original (antes de virar %) era cada calouro individual que respondeu à pesquisa dizendo sua área de interesse. São ~1,5 milhão de estudantes resumidos em 10 categorias.

In [ ]:
# (b)
hab = pd.read_excel('HABILITA.xls')
hab.Percentual.sum()

np.float64(99.9)

(b) A soma dos percentuais é 99,9%, e não exatamente 100%. Isso é consistente com o esperado — a diferença de 0,1 ponto percentual é apenas erro de arredondamento (cada percentual foi arredondado individualmente antes de somar). Não indica erro nos dados.

In [ ]:
# (c)
alt.Chart(hab).mark_bar().encode(
    x=alt.X('Percentual:Q'),
    y=alt.Y('Disciplina:N', sort='-x')
)

alt.Chart(...)

(c) Administração é a área mais procurada (16,8%), seguida de perto por Profissional (13,8%) e Artes&Humanidades (13,5%). As áreas menos procuradas são Ciências Físicas e cursos Técnicos, somando menos de 5% dos calouros. Não há uma área muito dominante — a distribuição é relativamente espalhada entre as 10 categorias.

# Questão 3

### Você ouve rádio?

O serviço de classificação Arbitron coloca as estações de rádio norte-americanas em mais de 50 categorias que descrevem o tipo de programa que transmitem. Quais formatos atraem maior audiência?

No arquivo FORMATORADIO.xls estão as medidas dos níveis de audiência (pessoas com 12 anos ou mais) em um dados intante do tempo, para os formatos mais populares.

- (a) Qual a soma dos níveis de audiência para esses formatos? Qual o percentual da audiência de rádio que ouve estações em outros formatos?
- (b) Seria correto apresentar esses dados em um gráfico de setores? Por quê?
- (c) Faça um gráfico de barras para apresentar esses dados. Certifique-se de incluir uma categoria "Outro formato".







In [ ]:
# (a)
rad = pd.read_excel('FORMATORADIO.xls')

soma = rad.NivelAudiencia.sum()
outros = 100 - soma

print('Soma dos níveis de audiência:', soma)
print('Audiência de "outros formatos":', outros)

Soma dos níveis de audiência: 67.3
Audiência de "outros formatos": 32.7


(a) 32,7% da audiência de rádio ouve formatos que não estão entre os 12 listados (formatos "menores", cada um com pouca audiência individual).

(b) Não é ideal, pois há muitas categorias com valores próximos entre si, o que dificulta a comparação visual em um gráfico de setores. Um gráfico de barras é mais adequado para comparar esse tipo de valor.

In [ ]:
# (c)
# criar a linha "Outro formato" com o que falta pra 100%
rad.loc[len(rad)] = ['Outro formato', 100 - rad.NivelAudiencia.sum()]

alt.Chart(rad).mark_bar().encode(
    x='NivelAudiencia:Q',
    y=alt.Y('Formato:N', sort='-x')
)

alt.Chart(...)

(c) Isso adiciona uma barra "Outro formato" = 32,7% junto das outras 12, formando os 100%.

# Questão 4

### Nascidos Fora

Como os residentes nascidos fora dos Estados Unidos estão distribuídos?

O país, como um todo, tem 13% de residentes nascidos fora, mas nos estados esses percentuais variam de 1.2% na Virgínia Ocidental a 27.2% na Califórnia.

O arquivo NACIDOSFORA.xls apresenta os dados para os 51 estados.

- (a) Faça um histograma e analise brevemente seu resultado. Use classes de largura 5% começando em 0%. Isto é, a primeira barra cobre de 0% a < 5%, a segunda cobre de 5% a < 10%, e assim por diante.
- (b) Dê o resumo dos 5 números da distribuição.
- (c) A Califórnia é um potencial valor atípico ou simplesmente a maior observação da distribuição? O que diz a regra 1.5xAIQ?






In [ ]:
# (a)
nf = pd.read_excel('NASCIDOSFORA.xls')

alt.Chart(nf).mark_bar().encode(
    alt.X('PctNascFora:Q', bin=alt.Bin(step=5, extent=[0, 30])),
    y='count()'
)

alt.Chart(...)

(a) A distribuição é assimétrica à direita (unimodal): a maior parte dos estados concentra-se nas classes baixas (0% a 10%), com uma cauda longa de poucos estados com percentuais bem mais altos, indicando a presença de valores atípicos.

In [ ]:
# (b)
nf.PctNascFora.describe()

,PctNascFora
count,51.000000
mean,8.401961
std,6.053908
min,1.200000
25%,3.800000
50%,6.300000
75%,12.500000
max,27.200000


In [ ]:
# (c)
q1, q3 = nf.PctNascFora.quantile(0.25), nf.PctNascFora.quantile(0.75)
aiq = q3 - q1                      # AIQ = 12.5 - 3.8 = 8.7
limite_alto = q3 + 1.5 * aiq       # 12.5 + 1.5*8.7 = 25.55

print('Q1:', q1, '| Q3:', q3, '| AIQ:', aiq)
print('Limite superior (Q3 + 1.5*AIQ):', limite_alto)

Q1: 3.8 | Q3: 12.5 | AIQ: 8.7
Limite superior (Q3 + 1.5*AIQ): 25.549999999999997


(c) Pela regra do 1,5×AIQ, qualquer valor acima de 25,55% é considerado atípico. Como a Califórnia tem 27,2% — acima desse limite — ela é, sim, classificada como um valor atípico, e não apenas a maior observação 'normal' da distribuição.

# Questão 5

### Custo Faculdade

O arquivo FACCUSTO.xls possui os dados médios sobre mensalidades e taxas de matrícula cobradas de alunos do estado por faculdades e universidades de 4 anos, para os anos acadêmicos de 1980 a 2010. Os valores são dados em "dólares constantes", ajustados para teremo mesmo poder de compra de um dólar em 2010.

- (a) Faça um gráfico temporal das mensalidades e taxas médias.
- (b) Que padrão geral seu gráfico apresenta
- (c) Alguns possíveis desvios do padrão geral são valores atípicos, períodos em que as taxas caíram (em dólares de 2013), e períodos de aumentos particularmente rápidos. Quais estão presentes em seu gráfico, e durante quais anos?
- (d) Ao procurar por padrões, você acha que seria melhor o estudo de uma série temporal das taxas para cada ano ou o percentual de aumento para cada ano? Por quê?





In [ ]:
# (a)
fc = pd.read_excel('FACCUSTO.xls')

alt.Chart(fc).mark_line().encode(
    x='ano:O',
    y='mensalidade:Q'
)

alt.Chart(...)

(b) A mensalidade sobe de forma praticamente contínua de 1980 (US$ 2.119) até 2010 (US$ 7.605) — mais que triplicou em 30 anos, mesmo já ajustada pela inflação (dólares constantes de 2010).

In [ ]:
# (c)
fc['pct_aumento'] = fc['mensalidade'].pct_change() * 100
fc.sort_values('pct_aumento')

,ano,mensalidade,pct_aumento
20,2000,4426,0.659541
28,2008,6532,0.802469
7,1987,2845,1.065719
15,1995,4019,1.132360
19,1999,4397,1.360074
26,2006,6218,1.468668
8,1988,2903,2.038664
1,1981,2163,2.076451
17,1997,4226,2.299685
9,1989,2972,2.376852


(c) Não há períodos de queda. Há, porém, um período de aumento particularmente rápido em 2003 (+11%), possivelmente ligado a algum corte de verba pública ou crise orçamentária das universidades naquele ano.

(d) O percentual de aumento ano a ano é mais útil para identificar padrões de mudança (ex: anos de aceleração ou desaceleração), já que a série em nível (dólares) sempre sobe e disfarça essas variações. A série em valores absolutos, por outro lado, é melhor para mostrar o tamanho real do custo suportado pelas famílias. O ideal é usar as duas juntas: uma para 'quanto custa', outra para 'quão rápido está mudando'.

# Questão 6

### Emissão CO2

A queima de combustíveis em fábricas ou nos motores dos veiculos emite dióxido de carbono (CO2), o que contribui para o aquecimento global.

O arquivo EMISSAOCO2.xls apresenta as emissões anuais de CO2, por pessoa em 2010 em paises com populações de, no mínimo, 30 milhões naquele ano.

- (a) Por que você acha que escolhemos medir emissões por pessoa em vez do total de emissões de CO2 para cada país?
- (b) Faça um histograma a partir desses dados. Descreva a forma, o centro e a dispersão da distribuição.
- (c) Quais países são valores atípicos?
- (d) Ache a média e a mediana para esses dados. Quais características da distribuição (descrita no item b) explicam a diferença entre a média e a mediana?

(a) Medir o total de CO2 favoreceria automaticamente os países mais populosos (como China ou Índia), mesmo que cada pessoa individualmente polua pouco. Medir per capita permite comparar o impacto médio de cada pessoa, que é uma medida mais justa do padrão de consumo/estilo de vida de cada país, independente do tamanho da população.

In [ ]:
# (b)
co2 = pd.read_excel('EMISSAOCO2.xls')

alt.Chart(co2).mark_bar().encode(
    alt.X('CO2:Q', bin=alt.Bin(step=2)),
    y='count()'
)

alt.Chart(...)

(b)
- Forma: assimétrica à direita (unimodal), com a maioria dos países emitindo pouco e uma cauda de poucos países emitindo muito.

- Centro: mediana ≈ 3,95 toneladas/pessoa (média é maior: 4,61, puxada pelos valores altos).

- Dispersão: vai de 0,03 até 18,9 toneladas — bem espalhado (desvio padrão ≈ 4,73).

In [ ]:
# (c)
q1, q3 = co2.CO2.quantile(.25), co2.CO2.quantile(.75)
limite_alto = q3 + 1.5*(q3-q1)

print('Q1:', q1, '| Q3:', q3)
print('Limite superior:', limite_alto)

Q1: 0.7221 | Q3: 7.9239
Limite superior: 18.726599999999998


(c) Estados Unidos (18,91) fica acima do limite de 18,73 → é valor atípico. Canadá (16,92) fica logo abaixo do limite, então não é considerado atípico pela regra, mas está bem próximo/alto.

(d) Média = 4,61 | Mediana = 3,95 → a média é maior que a mediana.

Isso acontece porque a distribuição é assimétrica à direita: os poucos países com emissões muito altas (EUA, Canadá) 'puxam' a média para cima, enquanto a mediana — que é resistente a valores extremos — continua representando melhor o país 'típico'.

# Questão 7

### Tempo viagem ao trabalho na Carolina do Norte e Tempo de viagem ao trabalho em NY

A Pesquisa da Comunidade Americana pergunta, dentre muitas outras coisas, o tempo de viagem para o trabalho. No arquivo TEMPOVIAGEMCN.xls temos o tempo de viagem para 15 trabalhadores da Carolina do Norte, escolhidos aleatoriamente pelo Census Bureau, e no arquivo TEMPOVIAGEMNY.xls temos o tempo de viagem para o trabalho de 20 trabalhadores do estado de Nova York, escolhidos aleatoriamente.

Faça uma análise completa desses dados a partir da média, moda, mediana, quartis, boxplot e valores atípicos.

Não se esqueça de interpretar os resultados e também fazer análises comparativas entre as duas bases de dados.

In [ ]:
cn = pd.read_excel('TEMPOVIAGEMCN.xls')
ny = pd.read_excel('TEMPOVIAGEMNY.xls')

for nome, df in [('CN', cn), ('NY', ny)]:
    print(nome, df.Minutos.describe())
    print(df.Minutos.value_counts())

CN count    15.000000
mean     22.466667
std      15.230921
min       5.000000
25%      10.000000
50%      20.000000
75%      30.000000
max      60.000000
Name: Minutos, dtype: float64
Minutos
10    4
30    2
20    2
40    2
25    1
60    1
15    1
5     1
12    1
Name: count, dtype: int64
NY count    20.000000
mean     31.250000
std      21.877349
min       5.000000
25%      15.000000
50%      22.500000
75%      41.250000
max      85.000000
Name: Minutos, dtype: float64
Minutos
15    4
20    3
10    2
40    2
30    2
60    2
5     1
25    1
85    1
65    1
45    1
Name: count, dtype: int64


In [ ]:
cn['Origem'] = 'Carolina do Norte'
ny['Origem'] = 'Nova York'
juntos = pd.concat([cn, ny])

alt.Chart(juntos).mark_boxplot(extent=1.5).encode(
    x='Minutos:Q',
    y='Origem:N'
)

alt.Chart(...)

Nova York tem tempos de viagem consistentemente maiores (média e mediana mais altas) e também mais variabilidade (AIQ e desvio padrão maiores) do que a Carolina do Norte — o que faz sentido, já que NY tem trânsito urbano mais pesado e distâncias maiores em áreas metropolitanas. Além disso, NY apresenta um valor atípico (85 minutos), sugerindo pelo menos um trabalhador com deslocamento excepcionalmente longo, enquanto a Carolina do Norte não apresenta nenhum valor fora do padrão.

# Questão 8

### Estados Felizes

A felicidade humana, ou bem-estar, pode ser avaliada subjetivamente ou objetivamente. Uma avaliação subjetiva pode ser feita, ouvindo-se o que as pessoas dizem. Uma avaliação objetiva pode ser feita a partir de dados relacionados a bem-estar, como renda, preços das casas, ausência de congestionamentos de tráfego etc. As avaliações subjetivas e objetivas coincidem? Para estudar esse assunto, os investigadores fizeram avaliações, tanto subjetivas como objetivas, para cada um dos 50 estados norte-americanos (que constam no arquivo FELICIDADE.xls).

A medida subjetiva foi o escore médio de questões de satisfação com a vida, encontradas no Behavioral Risk Factor Surveillance System (BRFSS) (Sistema de Avaliação de Risco Comportamental), que é um sistema de pesquisas de saúde com base nos estados. Menores escores indicam um maior grau de felicidade. Para a avaliação objetiva da felicidade, os investigadores calcularam um escore de bem-estar (chamado escore de compensação de diferenciais) para cada estado, com base em medidas objetivas relacionadas à felicidade e ao bem-estar, Os estados foram, então, ordenados de acordo com esse escore (Posto 1 sendo o mais feliz).

- (a) Faça um diagrama de dispersão do BRFSS médio (variável resposta) contra o posto com base nos escores de compensação de diferenciais (variável explicativa).
- (b) Há uma associação geral positiva ou uma associação geral negativa entre BRFSS e o posto com base no método de compensação de diferenciais?
- (c) A associação geral mostra concordância ou discordância entre o escore médio subjetivo BRFSS e o posto com base nos dados objetivos usados no método de compensação de diferenciais?
- (d) Há algum valor atípico? Se for o caso, quais são os escores BRFSS correspondentes a esses valores atípicos?

In [ ]:
# (a)
fel = pd.read_excel('FELICIDADE.xls')

alt.Chart(fel).mark_point().encode(
    x=alt.X('CompDif:Q', title='Posto (compensação de diferenciais)'),  # variável explicativa
    y=alt.Y('BRFSS:Q', title='Escore médio BRFSS')                       # variável resposta
)

alt.Chart(...)

In [ ]:
fel.BRFSS.corr(fel.CompDif)

np.float64(-0.5644800727600424)

(b) correlação = -0,56 → associação negativa (moderada).

(c) Há discordância: estados que se saem melhor na avaliação objetiva (bem-estar material) tendem a reportar menor felicidade subjetiva, e vice-versa. Isso é contraintuitivo — sugere que bem-estar material e felicidade percebida não caminham juntos nesses dados.

In [ ]:
# (d)
q1, q3 = fel.BRFSS.quantile(.25), fel.BRFSS.quantile(.75)
aiq = q3 - q1
limite_baixo = q1 - 1.5 * aiq
limite_alto = q3 + 1.5 * aiq

print('Limite inferior:', limite_baixo, '| Limite superior:', limite_alto)

# valores fora desse intervalo seriam os atípicos:
fel[(fel.BRFSS < limite_baixo) | (fel.BRFSS > limite_alto)]

Limite inferior: -0.140875 | Limite superior: 0.062125


,Estado,BRFSS,CompDif


(d) nenhum valor de BRFSS ultrapassa os limites (todos entre -0,088 e 0,033) → não há valores atípicos claros nessa variável pela regra do 1,5×AIQ

# Questão 9

### Parar de fumar

Um grande experimento aleatorizado foi realizado para avaliar a eficácia de Chantix para parar de fumar, comparado com bupropion (mais comumente conhecido como Wellbutrin ou Zyban) e um placebo. Chantix é diferente da maioria dos produtos para parar de fumar, pois tem como alvo os receptores de nicotina no cérebro, ataca-os e impede que a nicotina os alcance, enquanto o bupropion é um antidepressivo muito usado para ajudar as pessoas a pararem de fumar.

Fumantes com boa saúde geral que fumavam, pelo menos, 10 cigarros por dia, foram associados aleatoriamente aos trata- mentos com Chantix (n = 352), com bupropion (n = 329), ou com o placebo (n = 344). A medida da resposta é a cessação contínua do fumo para as semanas 9 a 12 do estudo.

O arquivo CESSASUMO.xls é uma tabela de dupla entrada dos resultados.

Se um sujeito fumou nas semanas 9 a 12, como isso depende do tratamento recebido?

In [ ]:
ces = pd.read_excel('CESSAFUMO.xls')

for tratamento in ['Chantix', 'Bupropion', 'Placebo']:
    grupo = ces[ces.Tratamento == tratamento]           # filtra só esse tratamento
    total = grupo.Contagem.sum()                        # total de pessoas nesse grupo
    continuou_fumando = grupo[grupo.Fumou == 'Sim'].Contagem.sum()  # quantos continuaram fumando

    pct = continuou_fumando / total * 100
    print(tratamento, '-> % que continuou fumando:', pct)

Chantix -> % que continuou fumando: 55.96590909090909
Bupropion -> % que continuou fumando: 70.51671732522796
Placebo -> % que continuou fumando: 82.26744186046511


obs: a coluna Fumou = Sim significa que a pessoa continuou fumando nas semanas 9-12 (ou seja, o tratamento não funcionou para ela). Fumou = Não significa que ela parou de fumar (tratamento funcionou). Sempre confira os valores da coluna antes de interpretar — é fácil confundir "Sim" com "sucesso".

Há uma associação clara entre tratamento e desfecho: quem tomou Chantix teve a menor taxa de continuar fumando (55,97%), seguido por Bupropion (70,52%), enquanto o grupo Placebo teve a maior taxa de recaída (82,27%). Isso indica que Chantix foi o tratamento mais eficaz para parar de fumar entre os três, seguido por Bupropion, e o Placebo teve o pior desempenho — como era esperado, já que ele não tem princípio ativo algum.

# Questão 10

### Escores SAT Estaduais

O arquivo SATMAT.xls possui informações, para cada estado norte-americano, sobre o percentual de egressos do Ensino Médio que realizam o teste SAT para entrar em uma universidade e o escore médio estadual em matemática.

O percentual de estudantes do ensino médio que fazem o SAT varia de estado para estado. Esse fato ajuda a explicar as diferenças entre os estados relativas à média do escore SAT de matemática?

- (a) Faça uma análise unidimensional de ambas as variáveis com sua distribuição, média, desvio padrão e resumo dos 5 números
- (b) Faça uma análise bidimensional das variáveis (correlação entre elas), incluindo seu diagrama de dispersão e interpretação respondendo à pergunta levantada no enunciado da questão.

In [ ]:
# (a)
sat = pd.read_excel('SATMAT.xls')
sat.describe()


,PctSAT,SAT-Mat
count,51.000000,51.000000
mean,39.333333,538.058824
std,31.122768,39.245592
min,3.000000,462.000000
25%,8.000000,507.000000
50%,32.000000,526.000000
75%,68.500000,568.000000
max,100.000000,613.000000


(a)
- PctSAT: distribuição bem espalhada e assimétrica (vai de 3% a 100%).
- SAT-Mat: distribuição mais concentrada, entre 462 e 613 pontos.

In [ ]:
# (b)
# Célula 1: o gráfico de dispersão
alt.Chart(sat).mark_point().encode(
    x='PctSAT:Q',
    y='SAT-Mat:Q'
)

alt.Chart(...)

In [ ]:
# Célula 2: a correlação (o número)
sat['PctSAT'].corr(sat['SAT-Mat'])

np.float64(-0.8657710709847528)

(b) Sim, o percentual de alunos que fazem o SAT explica boa parte da diferença entre os estados: quanto maior o percentual de estudantes que fazem a prova, menor tende a ser a média estadual. Isso faz sentido: em estados onde poucos alunos fazem o SAT (ex: Dakota do Sul, 3%), geralmente só os alunos mais preparados/motivados (que pretendem cursar faculdades competitivas) fazem a prova, elevando a média. Já em estados onde o SAT é feito por quase todo mundo (ex: Maine, 100%), a média reflete o desempenho de um grupo bem mais heterogêneo (incluindo alunos menos preparados), puxando a média para baixo. Ou seja, a diferença não é necessariamente porque um estado 'ensina melhor' — é um efeito do grupo que faz a prova.

# Questão 11

### Índice de Desenvolvimento da Educação Básica

O Indice de Desenvolvimento da Educação Básica (Ideb) foi criado em 2007 com a união de dois indicadores: o fluxo escolar e a nota padronizada em testes de desempenho. Tecnicamente, o Ideb é estimado com base em informações secundárias do Censo Escolar, do Sistema de Avaliação da Educação Básica (Saeb) e da Prova Brasil no site do Instituto Nacional de Estudos e Pesquisas Educacionais Anisio Teixeira (Inep).

Identifique o valor do Ideb no seu estado e indique a
posição que ele ocupa no ranking nacional.

[Análise do índice pelo Poder 360](https://www.poder360.com.br/educacao/conheca-os-estados-com-as-notas-mais-altas-e-mais-baixas-no-ideb/)

[Para aprofundar: resultados Ideb 2023](https://www.gov.br/inep/pt-br/areas-de-atuacao/pesquisas-estatisticas-e-indicadores/ideb/resultados)

# Questão 12

### Análise de renda

Considere os dados do arquivo InfoFamiliasEntrevistadas.csv, corresponde ao resultado de entrevistas realizadas com uma amostra de 120 famílias de uma certa localidade.

As variáveis que compõem a base são:
- Local - localidade da moradia
- P.a.p. - uso de algum programa de alimentação popular (0 - não; 1 - sim)
- Instr. - último nível de instrução completado pela pessoa com principal renda da casa (1 - nenhum; 2 - fundamental; 3 - médio)
- Tam. - número de pessoas residentes no domicílio
- Renda - renda familiar mensal, em quantidade de salários mínimos



Responda às questões abaixo:

- (a) Classifique as famílias com renda mensal de até 5 salários mínimos, como _renda baixa_ e famílias com rendimentos mensais acima de 5 salários mínimos como _renda alta_.
- (b) A amostra sugere alguma associação entre _renda familiar_ e _uso de programas de alimentação popular_ ? Justifique através da construção e interpretação de uma tabela de dupla entrada.

In [ ]:
# (a)
fam = pd.read_csv('InfoFamiliasEntrevistadas.csv')

fam['FaixaRenda'] = 'Renda alta'                          # todo mundo começa como "alta"
fam.loc[fam.Renda <= 5, 'FaixaRenda'] = 'Renda baixa'      # troca pra "baixa" quem tem Renda <= 5

fam.FaixaRenda.value_counts()

,count
FaixaRenda,
Renda alta,65
Renda baixa,55


In [ ]:
# (b)
baixa = fam[fam.FaixaRenda == 'Renda baixa']
alta = fam[fam.FaixaRenda == 'Renda alta']

pct_pap_baixa = baixa[baixa['P.a.p.'] == 1].shape[0] / baixa.shape[0] * 100
pct_pap_alta = alta[alta['P.a.p.'] == 1].shape[0] / alta.shape[0] * 100

print('Renda baixa - % que usa PAP:', pct_pap_baixa)
print('Renda alta - % que usa PAP:', pct_pap_alta)

Renda baixa - % que usa PAP: 72.72727272727273
Renda alta - % que usa PAP: 58.46153846153847


(b) Sim, a amostra sugere associação: famílias de renda baixa usam o programa de alimentação popular proporcionalmente mais (72,7%) do que famílias de renda alta (58,5%). Isso é coerente com o esperado, já que programas de alimentação popular são voltados justamente para famílias de menor renda. A diferença de ~14 pontos percentuais indica uma associação moderada entre as duas variáveis — embora ambos os grupos usem o programa em proporção relativamente alta.

# Questão 13

### IDH OECD

A OECD (Organization for Economic Cooperation and Development – Organização para a Cooperação e Desenvolvimento Econômico) consiste em países desenvolvidos e industrializados que aceitam os princípios de democracia representativa e uma economia de mercado livre.

O arquivo OECD_IDH_data.csv mostra os dados das Nações Unidas para os países da OECD em várias variáveis: produto interno bruto (PIB, per capita em dólares americanos), percentual de desempregados, uma medida de desigualdade baseada na comparação da riqueza dos 10% mais ricos com os 10% mais pobres, gasto público em saúde (como um percentual do PIB), o número de médicos por 100000 pessoas, emissão de dióxido de carbono (CO2, per capita em toneladas métricas), o percentual de mulheres no parlamento e atividade econômica feminina como um percentual da taxa masculina.

- (a) Construa em histograma do PIB.
- (b)	Identifique o valor atípico no histograma.
- (c) Escolha outras duas variáveis para fazer a análise unidimensional da sua distribuição, interpretando o seu resultado.
- (d) Faça uma análise bidimensional das variáveis escolhidas em (c) e interprete essa relação.


In [ ]:
# (a)
oecd = pd.read_csv('OECD_IDH_data.csv')

alt.Chart(oecd).mark_bar().encode(
    alt.X('GDP:Q', bin=alt.Bin(step=5000)),
    y='count()'
)

alt.Chart(...)

(a) A maioria dos países fica entre 19 mil e 33 mil dólares per capita, com um valor bem isolado lá na ponta.

In [ ]:
# (b)
q1, q3 = oecd.GDP.quantile(.25), oecd.GDP.quantile(.75)
limite_alto = q3 + 1.5*(q3-q1)   # = 38.479,5

print('Q1:', q1, '| Q3:', q3)
print('Limite superior:', limite_alto)

Q1: 28777.0 | Q3: 32658.0
Limite superior: 38479.5


(b) Luxemburgo, com PIB per capita de US$ 69.961, está muito acima do limite (38.479,5) → é claramente um valor atípico.

In [ ]:
# (c)
oecd[['Health', 'FemEcon']].describe()

,Health,FemEcon
count,23.000000,23.000000
mean,6.717391,76.826087
std,1.067560,7.743925
min,5.100000,61.000000
25%,6.150000,72.000000
50%,6.400000,79.000000
75%,7.200000,82.000000
max,8.800000,87.000000


(c)
- Health (gasto público em saúde, % do PIB): média 6,7%, varia de 5,1% a 8,8%. Distribuição pouco espalhada.
- FemEcon (atividade econômica feminina, % da masculina): média 76,8%, varia de 61% a 87%.

In [ ]:
# (d)
# Célula 1: o número da correlação
print('Correlação:', oecd['Health'].corr(oecd['FemEcon']))

Correlação: 0.5842959553546753


In [ ]:
# Célula 2: o gráfico de dispersão
alt.Chart(oecd).mark_point().encode(x='Health:Q', y='FemEcon:Q')

alt.Chart(...)

(d) Países que investem proporcionalmente mais em saúde pública tendem também a ter maior participação econômica feminina relativa. A relação é positiva e moderada — não é uma regra fechada, mas sugere que investimento social mais amplo (saúde) caminha junto com maior equidade de gênero no mercado de trabalho nesses países.

# Questão 14

### Censo Demográfico dos Estados brasileiros em 2000

O arquivo Censo_estados.xlsx possui informações sobre o Cendo Demográfico de 2000, que contém inficadores sociais dos estados brasileiros.

Realize análises bivariadas e interprete-as, para cada par de variáveis.

In [ ]:
censo = pd.read_excel('Censo_estados.xlsx')

colunas = ['Esperança de vida ao nascer 2000', 'Mortalidade infantil 2000',
           'Taxa de analfabetismo - 18 anos ou mais de idade 2000', 'Renda per capita 2000']

censo[colunas].corr()   # gera a "matriz" com a correlação de cada par

,Esperança de vida ao nascer 2000,Mortalidade infantil 2000,Taxa de analfabetismo - 18 anos ou mais de idade 2000,Renda per capita 2000
Esperança de vida ao nascer 2000,1.000000,-0.886367,-0.854615,0.849826
Mortalidade infantil 2000,-0.886367,1.000000,0.930282,-0.822238
Taxa de analfabetismo - 18 anos ou mais de idade 2000,-0.854615,0.930282,1.000000,-0.832541
Renda per capita 2000,0.849826,-0.822238,-0.832541,1.000000


In [ ]:
alt.Chart(censo).mark_point().encode(
    x='Renda per capita 2000:Q',
    y='Esperança de vida ao nascer 2000:Q',
    tooltip=['Territorialidades']
)

alt.Chart(...)

Todos os pares mostram correlações fortes, no sentido esperado: estados com maior renda per capita têm maior esperança de vida, e menor mortalidade infantil e analfabetismo. Isso reflete o padrão clássico de desigualdade regional no Brasil em 2000 — indicadores sociais (saúde, educação) e econômicos (renda) caminham muito juntos: estados mais pobres tendem a apresentar piores indicadores em todas as outras dimensões simultaneamente.